### PR TITLE
[TOPIC-BLE-LLCP] EDTT integration

### PR DESCRIPTION
**Pull request by thoh-ot**
This PR adds:

- building both new and legacy variant of hci_test_app
- fine grained verbosity
- rr debug support

### CODE REVIEW

**Review by aescolar — COMMENTED** [review]
This seems overall ok. Most of these commits could target master I guess?

**Comment by thoh-ot** [comment]
> This seems overall ok. Most of these commits could target master I guess?

Yes. Only two commits are specific to the LLCP work.

**Review by erbr-ot — APPROVED** [review]
Looks sweet - nice!

File: tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/prj_legacy.conf [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+CONFIG_BT_CTLR_LE_ENC=y
+CONFIG_BT_CTLR_PRIVACY=y
+CONFIG_BT_CTLR_FILTER=y
+CONFIG_BT_CTLR_DTM_HCI=y
+CONFIG_BT_CTLR_TX_BUFFER_SIZE=60
+CONFIG_BT_CTLR_DATA_LENGTH_MAX=60
+
+CONFIG_SYS_POWER_MANAGEMENT=y
+CONFIG_TICKLESS_IDLE=y
+CONFIG_TICKLESS_KERNEL=y
```
Comment (kruithofa) [inline tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/prj_legacy.conf:27]: At some point in the future the default LL will switch to the refactored LLCP. Wouldn't it be better to explicitly select the legacy LLCP here:

`# Disabling BT_LL_SW_SPLIT_LLCP_LEGACY requires BT_CTLR_ADVANCED_FEATURES`
`CONFIG_BT_CTLR_ADVANCED_FEATURES=y`
`CONFIG_BT_LL_SW_SPLIT_LLCP_LEGACY=y`

Even though legacy is default now, when changing the default it is easy to overlook that this file needs to be changed at that time.

**Review by kruithofa — CHANGES_REQUESTED** [review]

**Comment by kruithofa** [comment]
It can be determined from looking at the scripts, but would it be worth adding a short explanation that PRJ_CONF_x can be set to specific project files, and that RR/RR_x can be used for invoking rr?

File: tests/bluetooth/bsim_bt/edtt_ble_test_app/tests_scripts/_controller_tests_inner.sh [hunk 2]
```diff
@@ -4,16 +4,32 @@
...
 
 function Execute(){
+  local rr=
+  if [ "rr" = "$1" ]; then
+    local devno=$2
+    shift 2
+    local exe=$(basename $1)
+    local out=/tmp/rr/$$-${SIMULATION_ID}-${exe}-d_${devno}
+    rm -rf ${out}
+    rr="rr record -o ${out}"
```
Comment (kruithofa) [inline tests/bluetooth/bsim_bt/edtt_ble_test_app/tests_scripts/_controller_tests_inner.sh:25]: if we are paranoid we could add a check to see if rr is installed

Comment (thoh-ot) [reply tests/bluetooth/bsim_bt/edtt_ble_test_app/tests_scripts/_controller_tests_inner.sh:25]: ACK

File: tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/prj_legacy.conf (hunk 1 above)
Comment (thoh-ot) [reply tests/bluetooth/bsim_bt/edtt_ble_test_app/hci_test_app/prj_legacy.conf:27]: ACK

**Comment by thoh-ot** [comment]
> It can be determined from looking at the scripts, but would it be worth adding a short explanation that PRJ_CONF_x can be set to specific project files, and that RR/RR_x can be used for invoking rr?

In the PR description or somewhere else?

**Review by kruithofa — APPROVED** [review]
LGTM

**Comment by kruithofa** [comment]
> > It can be determined from looking at the scripts, but would it be worth adding a short explanation that PRJ_CONF_x can be set to specific project files, and that RR/RR_x can be used for invoking rr?
> 
> In the PR description or somewhere else?

I think in the script itself would be best

**Comment by thoh-ot** [comment]
> > > It can be determined from looking at the scripts, but would it be worth adding a short explanation that PRJ_CONF_x can be set to specific project files, and that RR/RR_x can be used for invoking rr?
> > 
> > 
> > In the PR description or somewhere else?
> 
> I think in the script itself would be best

I have added some top level documentation of the environment variables in the script.

**Pull request merged**